java/src/main/java/com/google/crypto/tink/integration/awskms/AwsKmsClient.java

```diff
@@ -110,28 +110,29 @@ public KmsClient withDefaultCredentials() throws GeneralSecurityException {
   }
 
   /** Loads AWS credentials from a provider. */
-  public KmsClient withCredentialsProvider(AWSCredentialsProvider provider)
-      throws GeneralSecurityException {
-    try {
-      String[] tokens = this.keyUri.split(":");
-      this.client =
-          AWSKMSClientBuilder.standard()
-              .withCredentials(provider)
-              .withRegion(Regions.fromName(tokens[4]))
-              .build();
+    public KmsClient withCredentialsProvider(AWSCredentialsProvider provider)
+        throws GeneralSecurityException {
+      this.provider = provider;
```

Comment:
You forgot to declare provider as a private member of this class.

Reply:
fixed: 9a403c21